go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -272,15 +272,12 @@ func (c *addressManager) updateNodeAddressAnnotations() error {
 	return nil
 }
 
-func (c *addressManager) updateHostCIDRs(node *kapi.Node) error {
+func (c *addressManager) updateHostCIDRs(node *kapi.Node, ifAddrs []*net.IPNet) error {
 	if config.OvnKubeNode.Mode == types.NodeModeDPU {
 		// For DPU mode, here we need to use the DPU host's IP address which is the tenant cluster's
 		// host internal IP address instead.
-		nodeAddrStr, err := util.GetNodePrimaryIP(node)
-		if err != nil {
-			return err
-		}
-		nodeAddrSet := sets.New[string](nodeAddrStr)
+		nodeIPNetv4, _ := util.MatchFirstIPNetFamily(false, ifAddrs)
+		nodeAddrSet := sets.New[string](nodeIPNetv4.String())
```

Comment:
I would capture in a comment that we're intentionally only supporting IPv4 here. 

In general, how does `getDPUHostPrimaryIPAddresses` from https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_init.go#L218 relate to what we're doing here?

Reply:
Yep added the comment in. `getDPUHostPrimaryIPAddresses` could also been used for this.